### PR TITLE
Use join-contracts.py script for Etherscan verification

### LINKED ISSUE #525: Solidity source concatenation using join-contracts.py
This issue keeps track of using `join-contracts.py` from `etherscan_verify.py`.

Currently`etherscan_verify.py` uses string concatenation and string replacements to combine solidity codes into one file.  The code contains the names of all imported files and a specific pragma string:

https://github.com/raiden-network/raiden-contracts/blob/fdede381d1f205d404c67af988e2574df69eb1d9/raiden_contracts/deploy/etherscan_verify.py#L94

Before adding more contracts to `etherscan_verify.py`, I want to clean the script.

### PR DESCRIPTION
**Pull request by pirapira**
This pull-request removes the string concatenation in `etherscan_verify.py` and instead uses `join-contracts.py`.  This closes #525.

Some refactoring will follow (now possible because this PR removes contract-specific code).

Even after this change, Etherscan verification timeouts: https://github.com/raiden-network/raiden-contracts/issues/530 but at least I added an error message explaining how to get the source verified in a web browser.

### CODE REVIEW

File: raiden_contracts/deploy/etherscan_verify.py [hunk 1]
```diff
@@ -90,24 +92,23 @@ def etherscan_verify(chain_id, apikey, guid, contract_name):
...
+            'contracts/TokenNetworkRegistry.sol',
+            str(joined_file),
+        ]
+        old_working_dir = Path.cwd()
+        chdir(Path(__file__).parent.parent)
+        try:
+            combiner = subprocess.Popen(command)
+            combiner.wait()
+        finally:
+            chdir(old_working_dir)
```
Comment (karlb) [inline raiden_contracts/deploy/etherscan_verify.py]: You can avoid the `wait` call by using `subprocess.run` and the directory handling by using the `pwd` parameter:
```python
    subprocess.run(command, cwd=Path(__file__).parent.parent)
```

File: raiden_contracts/deploy/etherscan_verify.py [hunk 2]
```diff
@@ -174,8 +175,17 @@ def etherscan_verify_contract(
             status = r['status']
```
Comment (karlb) [inline raiden_contracts/deploy/etherscan_verify.py]: Not new in this PR, but a short comment what `content["status"] == "1"` means would be helpful.

File: raiden_contracts/deploy/etherscan_verify.py (hunk 1 above)
Comment (pirapira) [reply raiden_contracts/deploy/etherscan_verify.py]: I already touch this part in the next PR.

I didn't know `cwd` keyword.  It is really neat.  I'll use it there.

File: raiden_contracts/deploy/etherscan_verify.py (hunk 2 above)
Comment (pirapira) [reply raiden_contracts/deploy/etherscan_verify.py]: I added a comment.

**Pull request merged**